test/functional/admin/statistics_announcement_publications_controller_test.rb

```diff
@@ -0,0 +1,98 @@
+require "test_helper"
+
+class Admin::StatisticsAnnouncementPublicationsControllerTest < ActionController::TestCase
+  setup do
+    @user = login_as_preview_design_system_user(:gds_editor)
+    @announcement = create(:statistics_announcement)
+    @statistics_publication = create(:published_statistics)
+    @generic_publication = create(:publication)
+    @search = "publication-title"
+  end
+
+  should_be_an_admin_controller
+
+  view_test "GET :index with no search value renders search bar only" do
+    get :index, params: { statistics_announcement_id: @announcement }
+
+    assert_response :success
+    assert_select "input[name='search']"
+    refute_select ".govuk-table"
+  end
+
+  test "GET :index will only return statistical publications" do
+    Admin::EditionFilter.expects(:new).with([@statistics_publication], anything, anything)
+
+    get :index, params: { statistics_announcement_id: @announcement, search: @search }
+  end
+
+  test "GET :index will filter by title containing search param" do
+    create(:published_statistics, title: "something else")
+
+    Admin::EditionFilter.expects(:new).with([@statistics_publication], anything, anything)
+
+    get :index, params: { statistics_announcement_id: @announcement, search: @search }
+  end
+
+  view_test "GET :index with search value renders paginated results" do
+    editions = []
+    16.times { editions << @statistics_publication }
+
+    stub_filter = stub_edition_filter({ editions:, options: { per_page: 15 } })
+    Admin::EditionFilter.stubs(:new).returns(stub_filter)
```

Comment:
Hi @ollietreend, this is kinda where I started with the test, but found that the test user didn't have the necessary permissions to see the test publications, so they were being filtered out by the permitted_only method in the EditionFilter.

I picked up the stubbing from the Editions controller tests, which is probably better practice anyhow. Although, there doesn't seem to be a standalone test suite for the EditionFilter.

Reply:
Oh, weird. I would have thought `login_as_preview_design_system_user(:gds_editor)` would have permission to see those documents 🤷🏻 

I tend to prefer using factories (i.e. 'real' objects) rather than stubbing wherever possible, just because it gives a greater depth of test coverage by exercising more code. Plus it means the stub doesn't need updating if ever the public interface of the 'real' object changes.

But I don't have a strong opinion here. If `create_list(:published_statistics, 15)` isn't working, then feel free to stick with the stubs.

Reply:
Cool, thanks Ollie, I had another play with this, and still can't get the publications to render within the test. So will leave as is.